### PR TITLE
Gavin's work on read alignment

### PR DESCRIPTION
**Pull request by learner-long-life**
Gavin mentioned in Discord that his last work this spring when attempting to process the whole transcriptome into a suffix tree was on `read_alignment` branch.

During last week's Smarty Plants meeting, we discovered that the `main` method in `rusty-plants/main.rs` was commented out, as well as `algorithms/better_align.rs` so this started our exploration of how anyone of us can kickstart suffix tree building on our own machines.

This is the start of making a "Hello Plants" tutorial for anyone to try out Smarty Plants for themselves, it may be interesting for your work @EvergreenSpock .

### CODE REVIEW

**Comment by learner-long-life** [comment]
@AbyssalRemark has agreed to be the primary reviewer of rebasing @Gavin-Bowers `read_alignment` branch off of `main` , so that we can merge with the directory rename to `rusty-plants`. I may have borked something up, so please have a look and double-check.

I also installed `git-lfs` with Homebrew and ran `git lfs pull` in the repo to resolve `data/` large files.

It should be 9.9M lines in the full version as shown below. If you see it as 3 lines, that's just the ref in git LFS that needs to be downloaded.
```
 wc ../data/ref_genome.fna 
 9965732 9966398 807222515 ../data/ref_genome.fna
 ```

It built and ran, and is currently still running on my MacBook Pro with M1 (ARM).

image

image

File: rusty-plants/src/algorithms/read_align.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                index: start as usize - i,
+                errors: errors,
+            };
+            matches.push(new_match);
+        }
+    }
+    matches
+}
+
+// pub fn align_fragments(
```
Comment (AbyssalRemark) [inline rusty-plants/src/algorithms/read_align.rs:44]: This should just be deleted. if its needed its in the git history.

**Review by AbyssalRemark — APPROVED** [review]
Seems fine. I don't like this commenting out dead code but that's what happens when your try to measure contribution.

**Comment by learner-long-life** [comment]
> Seems fine. I don't like this commenting out dead code but that's what happens when your try to measure contribution.

GitHub counts both deletions and additions. We could talk about passing the same test with net code deletions as one way to show refactoring.

**Pull request merged**